### PR TITLE
Add Shutdown() method

### PR DESCRIPTION
**Pull request by polachok**

### CODE REVIEW

File: libvirt.go [hunk 1]
```diff
@@ -918,6 +933,39 @@ func (l *Libvirt) Version() (string, error) {
 	return versionString, nil
 }
 
+// Shutdown shuts down a domain
```
Comment (benlemasurier) [inline libvirt.go]: This comment can be improved and should be a full sentence. Add a note on the fact that if no flags are provided (`flags == 0`), libvirt will chose the best available method.

File: libvirttest/libvirt.go [hunk 2]
```diff
@@ -341,6 +341,17 @@ var testDefineXML = []byte{
...
+	0x00, 0x00, 0x00, 0x1c, // length
+	0x20, 0x00, 0x80, 0x86, // program
+	0x00, 0x00, 0x00, 0x01, // version
+	0x00, 0x00, 0x00, 0xea, // procedure
+	0x00, 0x00, 0x00, 0x01, // type
+	0x00, 0x00, 0x00, 0x00, // serial
+	0x00, 0x00, 0x00, 0x00, // status
+}
+
+
```
Comment (benlemasurier) [inline libvirttest/libvirt.go]: nit: this `\n`

File: libvirt_test.go [hunk 3]
```diff
@@ -370,3 +370,13 @@ func TestDefineXML(t *testing.T) {
...
 	}
 }
+
+func TestShutdown(t *testing.T) {
+	conn := libvirttest.New()
+	l := New(conn)
+
+	var flags ShutdownFlags
+	if err := l.Shutdown("test", flags); err != nil {
+		t.Fatalf("unexpected define error: %v", err)
```
Comment (benlemasurier) [inline libvirt_test.go]: did you mean "unexpected shutdown error"?

File: libvirt.go [hunk 4]
```diff
@@ -191,6 +191,21 @@ const (
 	DestroyFlagGraceful
 )
 
+// ShutdownFlags specifies options available when shutting down a domain.
+type ShutdownFlags uint32
+const (
+	// ShutdownAcpiPowerBtn - send ACPI event
+	ShutdownAcpiPowerBtn ShutdownFlags = 1 << iota
```
Comment (benlemasurier) [inline libvirt.go]: nit `\n` after each const definition here

Comment (polachok) [inline libvirt.go]: fixed

**Comment by benlemasurier** [comment]
Looks great! We've got a conflict from #35, fix that up and I'll get this merged.

**Comment by benlemasurier** [comment]
Looks great! We've got a conflict from #35, fix that up and I'll get this merged.

**Comment by polachok** [comment]
Done

**Comment by benlemasurier** [comment]
thank you!

**Pull request merged**